policy/modules/contrib/frr.te

```diff
@@ -0,0 +1,81 @@
+policy_module(frr, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type frr_t;
+type frr_exec_t;
+init_daemon_domain(frr_t, frr_exec_t)
+
+type frr_log_t;
+logging_log_file(frr_log_t)
+
+type frr_conf_t;
+files_config_file(frr_conf_t)
+
+type frr_unit_file_t;
+systemd_unit_file(frr_unit_file_t)
+
+type frr_var_run_t;
+files_pid_file(frr_var_run_t)
+
+########################################
+#
+# frr local policy
+#
+allow frr_t self:capability { dac_override kill net_bind_service net_raw setgid setuid };
```

Comment:
Did you investigate why frr needs the `dac_override` capability? Sometimes it's caused by wrong permissions being set on files.

Reply:
Yes, I investigate this issue, the frr need delete pid file **/var/run/frr/bgpd.pid** so need bypass ownership of the **rm** to delete this file. If you want I can send you full audit message.

Reply:
Looking at the daemons the service starts, it might indeed be unavoidable if the file is deleted by `watchfrr` (which runs as root) instead of `bgpd` itself. Please paste here the full audit record so that my curiosity is satisfied ;)

Reply:
Here you have :)
`type=PROCTITLE msg=audit(01/21/2021 09:15:43.987:358) : proctitle=rm -f /var/run/frr/bgpd.pid
type=PATH msg=audit(01/21/2021 09:15:43.987:358) : item=1 name=/var/run/frr/bgpd.pid inode=33039 dev=00:18 mode=file,644 ouid=frr ogid=frr rdev=00:00 obj=system_u:object_r:frr_var_run_t:s0 nametype=DELETE cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=PATH msg=audit(01/21/2021 09:15:43.987:358) : item=0 name=/var/run/frr/ inode=27604 dev=00:18 mode=dir,755 ouid=frr ogid=frr rdev=00:00 obj=system_u:object_r:frr_var_run_t:s0 nametype=PARENT cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0
type=CWD msg=audit(01/21/2021 09:15:43.987:358) : cwd=/
type=SYSCALL msg=audit(01/21/2021 09:15:43.987:358) : arch=x86_64 syscall=unlinkat success=yes exit=0 a0=0xffffff9c a1=0x55614f7d0640 a2=0x0 a3=0x0 items=2 ppid=4755 pid=4780 auid=unset uid=root gid=root euid=root suid=root fsuid=root egid=root sgid=root fsgid=root tty=(none) ses=unset comm=rm exe=/usr/bin/rm subj=system_u:system_r:frr_t:s0 key=(null)
type=AVC msg=audit(01/21/2021 09:15:43.987:358) : avc:  denied  { dac_override } for  pid=4780 comm=rm capability=dac_override  scontext=system_u:system_r:frr_t:s0 tcontext=system_u:system_r:frr_t:s0 tclass=capability permissive=1
`

Reply:
Yes, so it's a root process (presumably watchfrr) removing a pid file owned by user frr... in that case the dac_override is justified.